src/events_scratch_variable_create.js

```diff
@@ -3,9 +3,53 @@ import * as Blockly from "blockly/core";
 class ScratchVariableCreate extends Blockly.Events.VarCreate {
   constructor(variable) {
     super(variable);
+    if (!variable) return;
+
     this.isLocal = variable.isLocal;
     this.isCloud = variable.isCloud;
   }
+
+  toJson() {
+    const json = super.toJson();
+    json["isLocal"] = this.isLocal;
+    json["isCloud"] = this.isCloud;
+    return json;
+  }
+
+  static fromJson(json, workspace, event) {
+    const newEvent = super.fromJson(json, workspace, event);
+    newEvent.isLocal = json["isLocal"];
+    newEvent.isCloud = json["isCloud"];
+    return newEvent;
+  }
+
+  run(forward) {
+    const workspace = this.getEventWorkspace_();
+    if (forward) {
+      const VariableModel = Blockly.registry.getObject(
+        Blockly.registry.Type.VARIABLE_MODEL,
+        Blockly.registry.DEFAULT,
+        true
+      );
+      const variable = new VariableModel(
+        workspace,
+        this.varName,
+        this.varType,
+        this.varId,
+        this.isLocal,
+        this.isCloud
+      );
+      workspace.getVariableMap().addVariable(variable);
+      Blockly.Events.fire(
```

Comment:
We decided to have the load method fire a create event but not the constructor correct?

Reply:
What do you mean by "load method"? `addVariable`? I think it may makes sense to have that fire the event, because otherwise callers have to do so (as here).

Reply:
I mean the `load` method on the variable model.

Reply:
Ah got it, yes that's right.